File: tests/python/openai/test_openinference.py
"""Conformance test: OpenInference (Arize) OpenAI instrumentation."""

from common import run, run_chat, run_chat_streaming, run_chat_tool_call, run_embeddings


def instrument():
    from openinference.instrumentation.openai import OpenAIInstrumentor
    from openinference.instrumentation.config import TraceConfig

    OpenAIInstrumentor().instrument(config=TraceConfig(enable_genai_semconv=True))


if __name__ == "__main__":
    run(
        "OpenInference: OpenAI Conformance Test",
        instrument,
        [run_chat, run_chat_streaming, run_chat_tool_call, run_embeddings],
    )

File: tests/js/openai/test_openinference.ts
/**
 * Conformance test: OpenInference (Arize) JS OpenAI instrumentation.
 */

import { OpenAIInstrumentation } from "@arizeai/openinference-instrumentation-openai";
import { run } from "./common";

function instrument(openaiModule: any) {
  const instrumentation = new OpenAIInstrumentation();
  instrumentation.manuallyInstrument(openaiModule);
}

run("OpenInference JS: OpenAI Conformance Test", instrument).catch((e) => { console.error(e); process.exit(1); });
